### PR TITLE
[RFC] Add "Don't give the file editing message" flag in shortmess option.

### PR DESCRIPTION
**Pull request by Shougo**
It adds F flag in shortmess option.
It is "Don't give the file editing message" flag.
Because, it is useless if you use statusline feature.

Note:  I will create the patch for Vim after it is merged.

### CODE REVIEW

File: src/nvim/ex_cmds.c [hunk 1]
```diff
@@ -1506,8 +1506,11 @@ void ex_file(exarg_T *eap)
     if (rename_buffer(eap->arg) == FAIL)
       return;
   }
-  /* print full file name if :cd used */
-  fileinfo(FALSE, FALSE, eap->forceit);
+
+  if (!shortmess(SHM_FILEINFO)) {
+    /* print full file name if :cd used */
```
Comment (fmoralesc) [inline src/nvim/ex_cmds.c]: `\\` comment

File: src/nvim/ex_cmds.c [hunk 2]
```diff
@@ -1506,8 +1506,11 @@ void ex_file(exarg_T *eap)
...
+    fileinfo(FALSE, FALSE, eap->forceit);
```
Comment (fmoralesc) [inline src/nvim/ex_cmds.c]: `false`

File: src/nvim/ex_cmds.c [hunk 3]
```diff
@@ -2483,7 +2486,9 @@ do_ecmd (
     msg_scroll = msg_scroll_save;
     msg_scrolled_ign = TRUE;
 
-    fileinfo(FALSE, TRUE, FALSE);
+    if (!shortmess(SHM_FILEINFO)) {
+      fileinfo(FALSE, TRUE, FALSE);
```
Comment (fmoralesc) [inline src/nvim/ex_cmds.c]: `false`

**Comment by Shougo** [comment]
Thank you.  Fixed.

**Comment by wsdjeg** [comment]
does it mean ,when I start vim like ,`vim README.md`,then I see a message like `"README.md" 37L, 822C`
this flag is to hide this?

**Comment by Shougo** [comment]
> does it mean ,when I start vim like ,vim README.md,then I see a message like "README.md" 37L, 822C
> this flag is to hide this? 

Yes.

**Comment by wsdjeg** [comment]
I think I have search for `how to hide file info when open a new file` more than 20 times? thanks for this PR.

**Comment by Shougo** [comment]
> I think I have search for how to hide file info when open a new file more than 20 times?

I have did it.  But it is nothing.  So I have implemented.

**Comment by fmoralesc** [comment]
Is it possible to make tests for this, or does the message get lost in a redraw? I think this would be the only thing missing from this PR.

**Comment by Shougo** [comment]
> Is it possible to make tests for this, or does the message get lost in a redraw? I think this would be the only thing missing from this PR.

I don't know about the UI tests.

**Comment by fmoralesc** [comment]
I'm not an expert, but this seems enough:
~~~ lua
local helpers = require('test.functional.helpers')
local Screen = require('test.functional.ui.screen')
local clear, feed, execute = helpers.clear, helpers.feed, helpers.execute
                  
describe("'shortmess'", function()
  local screen

  before_each(function()
    clear()
    screen = Screen.new(25, 5)
    screen:attach()
    execute('set shortmess&')
  end)

  after_each(function()
    screen:detach()
  end)

  describe("=F", function()
    it(':hides messages about the files read', function()
      execute('e test')
      screen:expect([[
        ^                         |
        ~                        |
        ~                        |
        ~                        |
        "test" is a directory    |
      ]])
      execute('set shortmess=F')
      execute('e test')
      screen:expect([[
        ^                         |
        ~                        |
        ~                        |
        ~                        |
        :e test                  |
      ]])
    end)
  end)
end)
~~~
I created this in `test/functional/ui/shortmess_spec.lua`, but I'm not sure if it that is the place for it to go.

**Comment by Shougo** [comment]
Thank you!

**Comment by Shougo** [comment]
I have added the tests by @fmoralesc

**Comment by justinmk** [comment]
I suggest sending this to Vim first, so that we don't have to do double-work if Bram changes something. At least confirm that `F` is the flag he wants to use.

**Comment by Shougo** [comment]
OK.  I will do it.

**Comment by Shougo** [comment]
I have sent it to vim_dev.

https://github.com/vim/vim/pull/686

**Comment by Shougo** [comment]
The patch is merged in Vim 7.4.1570.
I have fixed the code.

**Comment by fmoralesc** [comment]
Looking good, and the vim compat detail is solved. :+1:

**Comment by fmoralesc** [comment]
@Shougo Just a couple fixes needed for appeasing clint.py. Other than that, this seems ready.

**Comment by Shougo** [comment]
I have fixed the code style.

**Comment by fmoralesc** [comment]
Excellent. Can you squash so we can merge it cleanly?

**Comment by Shougo** [comment]
I have squashed.

File: src/nvim/buffer.c [hunk 4]
```diff
@@ -143,10 +143,17 @@ open_buffer (
...
     retval = readfile(curbuf->b_ffname, curbuf->b_fname,
-        (linenr_T)0, (linenr_T)0, (linenr_T)MAXLNUM, eap,
-        flags | READ_NEW);
-    /* Help buffer is filtered. */
+                      (linenr_T)0, (linenr_T)0, (linenr_T)MAXLNUM, eap,
+                      flags | READ_NEW);
+    msg_silent = old_msg_silent;
+
+    // Help buffer is filtered.
     if (curbuf->b_help)
```
Comment (fmoralesc) [inline src/nvim/buffer.c]: clint.py is complaining about the lack of braces in this conditional. :(

File: runtime/doc/options.txt [hunk 5]
```diff
@@ -5539,6 +5539,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 		"-- XXX completion (YYY)", "match 1 of 2", "The only match",
 		"Pattern not found", "Back at original", etc.
 	  q     use "recording" instead of "recording @a"
+	  F     don't give the file info when editing a file, like `:silent`
+		was used for the command
```
Comment (justinmk) [inline runtime/doc/options.txt]: missing tabs after "q" and "F". (I'll fix this in merge)

File: test/functional/ui/shortmess_spec.lua [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+local clear, feed, execute = helpers.clear, helpers.feed, helpers.execute
+
+describe("'shortmess'", function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(25, 5)
+    screen:attach()
+    execute('set shortmess&')
```
Comment (justinmk) [inline test/functional/ui/shortmess_spec.lua]: `clear()` guarantees defaults for all options.

File: test/functional/ui/shortmess_spec.lua [hunk 7]
```diff
@@ -0,0 +1,40 @@
+local helpers = require('test.functional.helpers')
```
Comment (justinmk) [inline test/functional/ui/shortmess_spec.lua]: This test isn't testing UI features; it only happens to use `Screen` to test behavior of an option. I moved it to `test/functional/options`.

**Comment by justinmk** [comment]
Merged

**Comment by Shougo** [comment]
Thank you!

**Pull request closed without merging**